modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedRequest.java

```diff
@@ -129,18 +154,24 @@ private RatedRequest(String id, List<RatedDocument> ratedDocs, SearchSourceBuild
         this.summaryFields = new ArrayList<>();
     }
 
-    public RatedRequest(String id, List<RatedDocument> ratedDocs, Map<String, Object> params,
-            String templateId) {
-        this(id, ratedDocs, null, params, templateId);
-    }
-
-    public RatedRequest(String id, List<RatedDocument> ratedDocs, SearchSourceBuilder testRequest) {
-        this(id, ratedDocs, testRequest, new HashMap<>(), null);
+    static void validateEvaluatedQuery(SearchSourceBuilder evaluationRequest) {
+        // ensure that testRequest, if set, does not contain aggregation, suggest or highlighting section
+        if (evaluationRequest != null) {
```

Comment:
I wonder if `profile` and `explain` should be forbidden too?  Both have non-negligible impact on performance, and seem irrelevant to ranking as well.

Reply:
Good point, I think I will add that before I merge. Thanks.